File: momose/src/engine/HotSpot.java
package engine;


import java.awt.Color;
import java.util.Iterator;
import java.util.Vector;

import javax.media.opengl.GL;

import viewer.SimArenaGL;

import math.Point2d;


public class HotSpot extends ScenarioElement
 {
  private float radius;
  private int ID;
  private Vector connections;
  
  public HotSpot(Point2d pos,float radius,int ID)
   {
	super(pos);  
    this.radius=radius;
    this.ID=ID;
    connections=new Vector();
   }//Fine costruttore
  
  public Point2d getPosition()
   {return pos;}
 
  public float getRadius()
   {return radius;}
  
  public void setID(int ID)
   { this.ID=ID; }
 
  public int getID()
   { return ID; }
  
  public void createConnection(HotSpot hotSpot,float weight)
   { connections.add(new Connection(this,hotSpot,weight)); }
  
  public void createConnection(HotSpot hotSpot)
   { connections.add(new Connection(this,hotSpot)); }
  
  public void addConnection(Connection newConn)
   {connections.add(newConn);}
  
  public void addConnections(Vector newConns)
   {connections.addAll(newConns);}
  
  public Vector getConnections()
   {return connections;} 
  
  
  public void draw(GL gl, SimArenaGL arena) 
   {
	if(arena.getDrawHotSpot())
	 {  
	  //Salvo la matrice corrente
	  gl.glPushMatrix();
	  //Sposto il sistema di coordinate
	  gl.glTranslatef(pos.x,pos.y,0);
	  //Disegno il cerchio interno
	  arena.setActColor(gl,Color.LIGHT_GRAY);
	  gl.glScalef(radius,radius,0);
	  gl.glCallList(1);
	  //Riprendo la matrice corrente
	  gl.glPopMatrix();	
	  
	  //Stampo le connesioni di questo hotSpot
	 drawHotConnections(gl,arena);
	} 		
   }//Fine draw
  
private void drawHotConnections(GL gl,SimArenaGL arena)
  {
   Iterator i=connections.iterator();
   while(i.hasNext())
    {
     Connection nextConn=(Connection)i.next();
     HotSpot startSpot=nextConn.getStartSpot();
     HotSpot endSpot=nextConn.getEndSpot();
     arena.drawLine(gl,startSpot.getPosition(),endSpot.getPosition(),Color.LIGHT_GRAY);
    }   
  }//Fine drawHotConnections  
  
 
  public String toXml()
   {
	return new String(" <HotSpot ID=\""+ID+"\" x=\""+pos.x+"\" y=\""+pos.y+
			          "\" radius=\""+radius+"\"/>\n"); 
   }//Fine toXml

}//Fine classe HotSpot

File: momosecpp/src/HotSpot.cpp
#include "HotSpot.h"

HotSpot::HotSpot(const Point2D &pos,const float &radius,const int &ID):ScenarioElement(pos)
 {
  this->radius=radius;
  this->ID=ID;
 }//Fine Costruttore

HotSpot::~HotSpot(){}

void HotSpot::toXml(std::ostream &out)
 {
   out<<" <HotSpot ID=\""<<ID<<"\" x=\""<<p.x<<"\" y=\""<<p.y<<"\" radius=\""<<radius<<"\"/>\n";
 }//Fine toXml


///Metodi della classe Connection///
Connection::Connection(HotSpot *startSpot,HotSpot *endSpot)
 {
  this->startSpot=startSpot;  
  this->endSpot=endSpot;
  weight=0;	
 }//Fine costruttore
 
Connection::Connection(HotSpot *startSpot,HotSpot *endSpot,float weight)
 {
  this->startSpot=startSpot;  
  this->endSpot=endSpot;
  this->weight=weight;
 }//Fine costruttore

Connection::~Connection(){}


void HotSpot::addConnections(vector<Connection*> *newConns)
 {
  vector<Connection*>::iterator iConns;
  for(iConns=newConns->begin();iConns!=newConns->end();iConns++)
   {connections.push_back(*iConns);}
 }//Fine addConnections
 
void Connection::toXml(std::ostream &out)
 {
   out<<" <Connection startSpot=\""<<startSpot->getID()
      <<"\" endSpot=\""<<endSpot->getID()
	  <<"\" weight=\""<<weight<<"\"/>\n";
 }//Fine toXml
   



